General update to the role

wawrzek opened the pull request:
- Remove websever (it's Deprecated)
- Refactor role to follow new Ansible task syntax
- Rewrite Readme - put variables in alphabetic order and add short description for each
- Refactor variables names to make alphabetic order easier
- Remove pre Corda 3 node.conf and related tasks
- Add optional steps to register node with external Doorman
- Adjust corda user not to have sudo, shell and home directory

Commit:
Adjust to v3+ and remove webserver

Commit:
Remove more webserver calls

Commit:
Refactor tasks + add step to copy truststore

Commit:
Missing ':'

Commit:
One more missing ':'

Commit:
Fix typo

Commit:
One tab too far

Commit:
use loop

Commit:
Add extra step for initial-registration

Commit:
variable is string not bool

Commit:
Fix local_path var

Commit:
Fix 'when'

Commit:
Fix typo

Commit:
Add missing default variable

Commit:
Fix var name

Commit:
Fix access to file

Commit:
Ensure corda is started after registration

Commit:
Remove comments

Commit:
Add email address

Commit:
Adjust variables and update README

Commit:
Missed one variable rename

Commit:
Fix file access

Commit:
corda user doesn't need home

Commit:
Fix conditional for truststore copy

wzur-r3 commented on tasks/ubuntu.yml:
```diff
@@ -1,6 +1,8 @@
-  apt_key: keyserver=keyserver.ubuntu.com id=0x219BD9C9
+  apt_key:
+    keyserver: keyserver.ubuntu.com
+    id: 0x219BD9C9
```
This might be an old GPG key, see http://docs.azul.com/zulu/zuludocs/#ZuluUserGuide/PrepareZuluPlatform/AttachAPTRepositoryUbuntuOrDebianSys.htm

To avoid that problem in the future, this should be a variable with a default value of a current GPG key. Should it change in the future, at least that would be overwrite by a playbook, before this role is updated

wawrzek replied:
It worked, but in general good point. I'm going to create an 'issue' to fix it during next round of updates.

wawrzek replied:
Actually we might need to drop this all together following end of live for Java 8 (no free OracleJDK -> none OpenJDK updates)

wzur-r3 replied:
That particular GPG key *didn't* work for me when I started Vagrant boxes from example folder. That is how I found that issue.

wawrzek replied:
Worked for me - I know I used CentOS.

wawrzek replied:
fixed.

wzur-r3 commented on templates/node.conf.j2:
```diff
@@ -1,51 +1,49 @@
-    "dataSource.user" : sa
-    "dataSource.password" : ""
-}
+keyStorePassword = {{ corda_password_keystore }}
```
`corda_password_keystore` doesn't have any default value and Ansible might fail or walk over that, depending on the Ansible global settings.

It would be better to check if that value has been provided with `assert` Ansible module (see https://docs.ansible.com/ansible/latest/modules/assert_module.html) before using it, or to use `mandatory` (see https://docs.ansible.com/ansible/2.3/playbooks_filters.html#forcing-variables-to-be-defined)

wawrzek replied:
What do you suggest? I have no opinion.

wzur-r3 replied:
I would prefer `assert` as it could emit a meaningful message when that variable is not set

wawrzek replied:
Actually, there is a bigger problem. It was suggested not to set this variable, but pass it as environment variable. 
Another option is to use  Ansible to pass variable from vault to restart/start Corda. For now I'm going to use assert and log an issue.

wawrzek replied:
I used 'assert' and created new issue.

wzur-r3 reviewed (commented):
A general note, while it is good there is an example for Vagrant, it should be used as a general tool for testing roles

Commit:
Use assert for keystore password

- use assert to ensure that keystore password is set
- adjust 2 defaults

Commit:
Fix problem with repo key id

I think that actuall problem was that id wasn't string and ansible
translated it to decimal value

wzur-r3 commented on tasks/main.yml:
```diff
@@ -1,4 +1,9 @@
+- name: Check that password for keystore is defined
+  assert:
+    that: corda_password_keystore is defined
+    msg: "Having password to keystore in config file is not idea, but for now please set 'corda_password_keystore' to a value"
```
I believe there is a typo in `Having password to keystore in config file is not idea`

Also, even the source of `corda_password_keystore` can be anything, an environment variable, a variable from vault, and so on.
